Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -477,6 +477,56 @@ void TextureCacheBase::BindTextures()
   }
 }
 
+struct ManualMipmapDetector
+{
+
+  ManualMipmapDetector()
+  {
+    min = { 255, 255, 255, 255 };
+    max = { 0, 0, 0, 0 };
+  }
+
+  void AddLevel(u32 width, u32 height, u32 row_length, const u8* buffer)
+  {
+    ++levels;
+
+    u64 avg[4] = { 0 };
+    for (u32 i = 0; i < height; ++i)
+    {
+      auto* row = buffer;
+      for (u32 j = 0; j < width; ++j)
+      {
+        avg[0] += row[0];
+        avg[1] += row[1];
+        avg[2] += row[2];
+        avg[3] += row[3];
+
+        row += 4;
+      }
+      
+      buffer += row_length;
+    }
+
+    for (u32 c = 0; c < 4; ++c)
+    {
+      u8 cur = avg[c] / (width * height);
+      max[c] = std::max(cur, max[c]);
+      min[c] = std::min(cur, min[c]);
+    }
+  }
+
+  bool HasManualMipmaps() const {
+    const u8 THRESHOLD = 50;
```

Comment:
Oh wow, I didn't expect this to work at all with such a *huge* threshold. But as it does so, fine for me ;)

Reply:
Well it's not that high - it's 50 summed across 4 channels, so it's a mere 5% of divergence from the average (or more in a single channel of course).  Already at 40 I could see some normal mips failing the threshold, so actually I hoped it would be more obvious.